### PR TITLE
[msal-node]: Transform auth-code-pkce sample to typescript

### PR DESCRIPTION
**Pull request by samuelkubai**
With this change we transform the auth-code-pkce sample from Javascript to Typescript, in order for it to serve as an example for how to use msal-node in a typescript project.

### CODE REVIEW

File: samples/msal-node-samples/auth-code-pkce/src/index.ts [hunk 1]
```diff
@@ -2,32 +2,32 @@
...
+import session from "express-session";
+import { PublicClientApplication, AuthorizationCodeRequest, LogLevel, CryptoProvider } from "@azure/msal-node";
+import { RequestWithPKCE } from "./types";
 
 const SERVER_PORT = process.env.PORT || 3000;
 
-// Before running the sample, you will need to replace the values in the config, 
-// including the clientSecret
+// Before running the sample, you will need to replace the values in the config.
 const config = {
```
Comment (derisen) [inline samples/msal-node-samples/auth-code-pkce/src/index.ts:13]: `config` could be of type _Configuration_ from msal-node

File: samples/msal-node-samples/auth-code-pkce/src/index.ts [hunk 2]
```diff
@@ -99,14 +99,14 @@ app.get('/', (req, res) => {
     });
```
Comment (derisen) [inline samples/msal-node-samples/auth-code-pkce/src/index.ts:99]: maybe use `CommonAuthorizationUrlRequest` above?

```typescript
const authCodeUrlParameters: CommonAuthorizationUrlRequest = { /*...*/ }
```

File: samples/msal-node-samples/auth-code-pkce/src/types/index.ts [hunk 3]
```diff
@@ -0,0 +1,4 @@
+import { Request} from "express";
+import { Session } from "express-session";
+
+export type RequestWithPKCE = Request & { session: Session & { pkceCodes: { challengeMethod: string, challenge?: string, verifier?: string } }};
```
Comment (derisen) [inline samples/msal-node-samples/auth-code-pkce/src/types/index.ts:4]: Have you considered [declaration merging](https://www.typescriptlang.org/docs/handbook/declaration-merging.html) instead? Seems to be favored [in some situations](https://stackoverflow.com/questions/37377731/extend-express-request-object-using-typescript), but this is fine too.

**Review by derisen — APPROVED** [review]
LGTM! few minor comments

**Review by jasonnutter — APPROVED** [review]

**Pull request merged**